include/rocksdb/env.h

```diff
@@ -619,6 +619,9 @@ class Env {
   // Returns the ID of the current thread.
   virtual uint64_t GetThreadID() const;
 
+  // Returns the current process ID.
+  virtual int64_t GetProcessID() const = 0;
```

Comment:
Declaring it as pure virtual might break some code compilation. Can we provide a default implementation that calls a corresponding function in ```port/```?

Reply:
I'm just trying to follow the pattern, but unlike GetThreadID, GetProcessID does not have a platform-independent implementation via C++ APIs. I don't really understand the point of having functions like this in Env anyway. I've been criticized for not putting things with platform-specific implementations in Env that didn't have a specific need to be in the public API (criticized by @mrambacher re: GenerateRfcUuidFromPlatform in #8362).

Reply:
Agree about not having it in Env. I don't think anyone would have a reason to override it, which seems to be the main advantage of having it in Env. How about defining it in ```port_posix.cc``` (```getpid``` is Posix anyway) and ```port_win.cc```?

Reply:
@pdillinger I (mostly) agree about it not being in Env.  At the moment, the Env is the only container for port-specific values.  Since this method (along with ThreadID) seems unlikely to be overridden by anything, the only advantage to putting it in Env is if the function was going to be publicly exposed (since port/* is not in the pubic API).  In reality, most of the methods that are left as truly "Env" methods (and not in FileSystem or SystemClock) fall into that same pattern (where they will not be overridden by implementation).

A slightly different approach would be to make a "static Env::GetProcessID()" method and have that method call the appropriate port/* method.  This allows exposure without the concerns of virtual/override and implementations.